include/boost/gil/image_processing/morphology.hpp

```diff
@@ -0,0 +1,193 @@
+                            max_overlapped_element = src_view(col_boundary, row_boundary);
+                        if(src_view(col_boundary, row_boundary) < min_overlapped_element)
+                            min_overlapped_element = src_view(col_boundary, row_boundary);
+                    }
+                }
+            }
+            if(identifier)//identifier = 1 for dilation
+                dst_view(view_col, view_row) = max_overlapped_element;
+            else //identifier = 0 for erosion
+                dst_view(view_col, view_row) = min_overlapped_element; 
+        }
+        }
+    }
+
+template <typename SrcView, typename DstView,typename Kernel>
+void morph(SrcView const& src_view, DstView const& dst_view,Kernel const& ker_mat,const int identifier)
+{
+    using namespace boost::gil;
+    BOOST_ASSERT(ker_mat.size() != 0 && src_view.dimensions() == dst_view.dimensions());
+    gil::gray8_image_t intermediate_img(src_view.dimensions());
+    for (std::size_t i = 0; i < src_view.num_channels(); i++)
+    {
+        morph_impl(
+            nth_channel_view(src_view, i),
+            nth_channel_view(view(intermediate_img), i),
+            ker_mat,identifier
+        );
+    }
+    gil::transform_pixels(view(intermediate_img),dst_view,[](const gil::gray8_pixel_t& p){return p[0];});
+}
+
+//This function calculates the difference between pixel values of first image_view and second
+//image_view.This function can be used for rgb as well as grayscale images.
+template <typename SrcView, typename DstView ,typename DiffView>
+void difference_impl(SrcView const& src_view, DstView const& dst_view, DiffView const& diff_view)
+{
+    for (std::ptrdiff_t view_row = 0; view_row < src_view.height(); ++view_row)
+        for (std::ptrdiff_t view_col = 0; view_col < src_view.width(); ++view_col)
+            diff_view(view_col, view_row) = src_view(view_col,view_row) - dst_view(view_col,view_row);
+}
+
+template <typename SrcView, typename DstView, typename DiffView>
+void difference(SrcView const& src_view, DstView const& dst_view , DiffView const& diff_view)
+{
+
+    BOOST_ASSERT(src_view.dimensions() == dst_view.dimensions());
+
+    for (std::size_t i = 0; i < src_view.num_channels(); i++)
+    {
+        difference_impl(
+            nth_channel_view(src_view, i),
+            nth_channel_view(dst_view, i),
+            nth_channel_view(diff_view, i)
+        );
+    }
+}
+}//namespace boost::gil::detail
+
+// Dilation:Give the maximum overlapped value to the pixel overlapping with the center element of 
+//structuring element.We can vary the number of times dilation happens by varying the 
+// argument 'iterations' in the dilate function.
+template <typename SrcView, typename DstView,typename Kernel>
+void dilate(SrcView const& src_view, DstView const& dst_view,Kernel const& ker_mat,const int iterations)
+{
+    boost::gil::transform_pixels(src_view,dst_view,[](const boost::gil::gray8_pixel_t& p){return p[0];});
```

Comment:
I doubt it will work like this, leaving aside  that not every view will have `gray8_pixel_t` as pixel type. Can't nail down counterexample, but it will look like this:
```
input = [
    [max_value, std_value, std_value, std_value],
    [std_value] * 4,
    [std_value] * 4
]

kernel = [1] * 3 * 3
```

Please test dilation with the input above. `std_value` is something lower than `max_value`. I expect first 3x3 segment to be `max_value` and final column to be `std_value`.

Reply:
I tested this as ....
'''
input = [{ 255, 100, 100, 100},
        { 100, 100, 100, 100},
        { 100, 100, 100, 100}]

kernel = [{1,1,1},
               {1,1,1},
               {1,1,1}]

expected_output_for_dilation = [ { 255, 255, 100, 100},
                                                    { 255, 255, 100, 100},
                                                    { 100, 100, 100, 100}]
'''
This gave me no errors,can u please point out what exactly is breaking

Reply:
I am trying to find a solution for the first problem

Reply:
@simmplecoder ,shall i use this ``` copy_pixels(src_view,dst_view);```  to solve the first problem?

Reply:
@simmplecoder , I have included this example in test file, please take a look at it and tell me if I need to change anything.

Reply:
Looks good, you can resolve this thread (if it is resolvable).